### ISSUE TITLE
UX: Don't switch to the source (application, window, tab) you decide to record

### ISSUE THREAD

**Issue by oas777** [elan-ev/opencast-studio#403]
I think it's highly irritating that you switch to the source you select when choosing the desktop recording content. Stick in the original browser/tab and show the preview instead.

**Comment by LukasKalbertodt** [comment]
This is unfortunately not something we have control over. The browser brings the application/window/tab in the front/makes it active. I did not find a way for the web application to ask the browser not to do that. It might also be a security measure to make it more clear to the user that that window is now captured. 

So I'm afraid we probably have to close this as "can't fix this". (And -- not that it matters, but -- I agree with you in that the current behavior is kind of annoying.)

**Comment by oas777** [comment]
Stimmt, das ist mir in anderen Anwendungen so noch nicht aufgefallen. Schade. Wir müssen meines Erachtens die Nutzenden darauf hinweisen. Was wäre die letzte Stelle zwischen "What video source(s) to record?" und dem Wechsel in das dann gewählte Fenster/Tab/Programm, wo du eingreifen kannst? Da, wo "Display" steht oder auch im nachfolgenden Dialogfenster?

**Comment by LukasKalbertodt** [comment]
I [asked on StackOverflow](https://stackoverflow.com/q/60172910/2408867) about this again, but as expected, it doesn't seem like there is anything we can do.

Regarding telling the user: I'm not sure if that's really worth it or if we can properly do that. We basically would have to show this information once the users clicks on the "Display" or "Display&Camera" button. But how to show it? I don't really think there is a good way. 

I think in the long term we want a "help" page with an FAQ with all this information. There are a few tricky things when recording a window/your screen anyway. We can address those on that help page. Until then, I'll denote this to low priority for now.

**Comment by oas777** [comment]
I agree we need FAQ (or an explaining screencast...) in the long run. My question is whether we have to "warn" the user when selecting the content. Pleas find two suggestions attached, below is the respective text.
![tip-2](https://user-images.githubusercontent.com/1752342/74733835-76f5c880-524d-11ea-88a1-9377ba9df7dd.JPG)
![tip-1](https://user-images.githubusercontent.com/1752342/74733837-778e5f00-524d-11ea-8692-2b829c8420a0.JPG)

When selecting the display content, STUDIO will put forward that content for you to check; come back to STUDIO to continue the recording process.


Upon selection, STUDIO will put forward that content for you to check; come back to STUDIO to continue the recording process.

**Comment by LukasKalbertodt** [comment]
I opened an issue [on the `getDisplayMedia` standard repository](https://github.com/w3c/mediacapture-screen-share/issues/138). Hopefully people agree that this current behavior is not universally useful. However, even if people agree, it would take quite some time to make it into browsers.

In the meantime... mh. I saw your suggestions. Your first image is not possible as we don't have any control over the popup window. The second one would be possible, but I also don't think it's too great from a design/UX standpoint. Maybe #483 can help here...

**Comment by oas777** [comment]
Thanks, Lukas. Let's hope we can change that in the browsers used in the long run. As for a short-term solution, I see your concerns regarding design/UX/UI, though I doubt the explanation video alone will help. Would it be possible to to provide the information I proposed in the hover-over?

**Comment by oas777** [comment]
+1. I checked with other services (https://www.screencapture.com/, https://screenapp.io/#/), they have the same "issue". Maybe also worth looking at for their UI/UX. For example, I like the option to get other formats of a given recording

![image](https://github.com/elan-ev/opencast-studio/assets/1752342/a4ce14bf-8573-4998-8181-40c630f849ad)

which in our case would tell people to run this through Opencast.
